src/core/annotation.js

```diff
@@ -1985,23 +2027,101 @@ class WidgetAnnotation extends Annotation {
       );
     }
 
-    const font = await WidgetAnnotation._getFontData(
+    let font = await WidgetAnnotation._getFontData(
       evaluator,
       task,
       this.data.defaultAppearanceData,
       this._fieldResources.mergedResources
     );
-    const [defaultAppearance, fontSize] = this._computeFontSize(
-      totalHeight - defaultPadding,
-      totalWidth - 2 * hPadding,
-      value,
-      font,
-      lineCount
-    );
+
+    let defaultAppearance, fontSize, lineHeight;
+    const encodedLines = [];
+    let encodingError = false;
+    for (const line of lines) {
+      const encodedString = font.encodeString(line);
+      if (encodedString.length > 1) {
+        encodingError = true;
+      }
+      encodedLines.push(encodedString.join(""));
+    }
+
+    if (encodingError && intent & RenderingIntentFlag.SAVE) {
+      // We don't have a way to render the field, so we just rely on the
+      // /NeedAppearances trick to let the different sofware render correctly
```

Comment:
Generally speaking, thanks a lot for adding these comments since that definitely helps!

Here, and below, a few spelling/grammar nits:
```suggestion
      // /NeedAppearances trick to let the different software correctly render
```